uxarray/grid/neighbors.py

```diff
@@ -0,0 +1,220 @@
+        count_only : bool, default=False
+            if True, return only the count of points within distance r if False, return the indices of all points within distance r If return_distance==True, setting count_only=True will result in an error.
+        sort_results : bool, default=False
+           if True, the distances and indices will be sorted before being returned. If False, the results will not be sorted. If return_distance == False, setting sort_results = True will result in an error.
+
+
+        Returns
+        -------
+        d : ndarray of shape (xy.shape[0], k), dtype=double
+            Each entry gives the list of distances to the neighbors of the corresponding point.
+        ind : ndarray of shape (xy.shape[0], k), dtype=INT_DTYPE
+            Each entry gives the list of indices of neighbors of the corresponding point.
+        """
+
+        if r < 0.0:
+            raise AssertionError(
+                f"The value of r must be greater than or equal to zero.")
+
+        r = np.deg2rad(r)
+        xy = _prepare_xy_for_query(xy, use_radians)
+
+        if count_only:
+            count = self.tree.query_radius(xy, r, return_distance, count_only,
+                                           sort_results)
+
+            return count
+
+        if not count_only and not return_distance:
+            ind = self.tree.query_radius(xy, r, return_distance, count_only,
+                                         sort_results)
+
+            ind = np.asarray(ind, dtype=INT_DTYPE)
+
+            if xy.shape[0] == 1:
+                ind = ind.squeeze()
+
+            return ind
+
+        else:
+            ind, d = self.tree.query_radius(xy, r, return_distance, count_only,
+                                            sort_results)
+
+            ind = np.asarray(ind[0], dtype=INT_DTYPE)
+
+            if xy.shape[0] == 1:
+                ind = ind.squeeze()
+
+            d = np.rad2deg(d[0])
+
+            return d, ind
+
+
+def _prepare_xy_for_query(xy, use_radians):
+    """Prepares xy coordinates for query with the sklearn BallTree."""
+
+    xy = np.asarray(xy)
+
+    # expand if only a single node pair is provided
+    if xy.ndim == 1:
+        xy = np.expand_dims(xy, axis=0)
+
+    # expected shape is [n_pairs, 2]
+    if xy.shape[1] == 3:
+        raise AssertionError(
+            f"The dimension of each coordinate pair must be two (lon, lat). Did you attempt to query using Cartesian "
+            f"(x, y, z) coordinates?")
+
+    if xy.shape[1] != 2:
+        raise AssertionError(
+            f"The dimension of each coordinate pair must be two (lon, lat).)")
+
+    # swap X and Y for query
+    xy[:, [0, 1]] = xy[:, [1, 0]]
+
+    # balltree expects units in radians for query
+    if not use_radians:
+        xy = np.deg2rad(xy)
```

Comment:
Something should be wrong here: if not use_radians, then convert to radians?

Reply:
`use_radians` indicates whether the user is already providing their `xy` input in radians.